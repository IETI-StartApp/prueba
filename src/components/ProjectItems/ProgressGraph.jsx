import React from 'react';
import { Box, CircularProgress, Typography } from '@material-ui/core';
import { useStyles } from './styles';

export default function ProgressGraph( { projectId }){

    const classes = useStyles();
    const progress = 70;

    return (
        <Box position="relative" display="inline-flex">
            <CircularProgress variant="determinate" value={ progress } className={classes.mainColor} />
            <Box
                top={0}
                left={0}
                bottom={0}
                right={0}
                position="absolute"
                display="flex"
                alignItems="center"
                justifyContent="center"
            >
                <Typography variant="caption" component="div">
                    {
                        `${Math.round(progress,)}%`
                    }
                </Typography>
            </Box>
        </Box>
    );
}