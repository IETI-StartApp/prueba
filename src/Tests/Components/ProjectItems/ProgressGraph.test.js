import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import React from 'react';
import ProgressGraph from "../../../../src/components/ProjectItems/ProgressGraph";
import { Typography } from '@material-ui/core';

describe('Testing progress graph', () => {

    const wrapper = shallow(<ProgressGraph/>)

    test('should load ProgressGraph', () => {
        expect(wrapper).toMatchSnapshot();
    })

    test('should load tipography with a number before "%" ', () => {

        const progressPercentage = wrapper.find(Typography).at(0).text();
        expect(progressPercentage.length).toBeGreaterThanOrEqual(2);
        expect(progressPercentage.length).toBeLessThanOrEqual(4);
        expect (progressPercentage).toContain("%");

        const number = progressPercentage.substring(0, progressPercentage.length - 1);
        expect(number).not.toBeNaN();

    })

    test('should load tipography with a valid percentage', () => {
        const progressPercentage = wrapper.find(Typography).at(0).text();
        const number = progressPercentage.substring(0, progressPercentage.length - 1);
        expect(parseFloat(number)).toBeGreaterThanOrEqual(0);
        expect(parseFloat(number)).toBeLessThanOrEqual(100);
    })

})