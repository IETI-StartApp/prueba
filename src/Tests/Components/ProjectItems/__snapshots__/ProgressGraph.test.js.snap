// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing progress graph should load ProgressGraph 1`] = `
<Styled(MuiBox)
  display="inline-flex"
  position="relative"
>
  <WithStyles(ForwardRef(CircularProgress))
    className="makeStyles-mainColor-2"
    value={70}
    variant="determinate"
  />
  <Styled(MuiBox)
    alignItems="center"
    bottom={0}
    display="flex"
    justifyContent="center"
    left={0}
    position="absolute"
    right={0}
    top={0}
  >
    <WithStyles(ForwardRef(Typography))
      component="div"
      variant="caption"
    >
      70%
    </WithStyles(ForwardRef(Typography))>
  </Styled(MuiBox)>
</Styled(MuiBox)>
`;
